services/streammanager/kafka/kafkamanager.go

```diff
@@ -0,0 +1,178 @@
+package kafka
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"github.com/rudderlabs/rudder-server/config"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/tidwall/gjson"
+)
+
+// Config is the config that is required to send data to Kafka
+type Config struct {
+	Topic         string
+	HostName      string
+	Port          string
+	SslEnabled    bool
+	CACertificate string
+}
+
+var (
+	clientCertFile, clientKeyFile string
+	certificate                   tls.Certificate
+)
+
+func init() {
+	loadConfig()
+	loadCertificate()
+}
+
+func loadConfig() {
+	clientCertFile = config.GetEnv("SSL_CERTIFICATE_FILE_PATH", "")
+	clientKeyFile = config.GetEnv("SSL_KEY_FILE_PATH", "")
+}
+
+func loadCertificate() {
+	if clientCertFile != "" && clientKeyFile != "" {
+		cert, err := tls.LoadX509KeyPair(clientCertFile, clientKeyFile)
+		if err != nil {
+			logger.Errorf("error loading certificate: ", err)
+			return
+		}
+		certificate = cert
+	}
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destinationConfig interface{}) (sarama.SyncProducer, error) {
+
+	var destConfig = Config{}
+	jsonConfig, err := json.Marshal(destinationConfig)
+	err = json.Unmarshal(jsonConfig, &destConfig)
+
+	hostName := destConfig.HostName + ":" + destConfig.Port
+	isSslEnabled := destConfig.SslEnabled
+
+	hosts := []string{hostName}
+
+	config := sarama.NewConfig()
+	config.Producer.Partitioner = sarama.NewReferenceHashPartitioner
+	config.Producer.RequiredAcks = sarama.WaitForAll
```

Comment:
We need to put this in docs (all the values we are using by default). If someone wants to override, we should ask them to contact us, so we can understand their case better.